tests/unit/test_asyn.py

```diff
@@ -161,3 +161,31 @@ async def process(input):
                 list(mapper.iterate(timeout=4))
         finally:
             join_all_tasks(loop)
+
+
+def test_iter_over_async():
+    async def gen():
+        yield 1
+        yield 2
+
+    res = iter_over_async(gen(), get_loop())
+    assert list(res) == [1, 2]
+
+
+def test_iter_over_async_exception():
+    async def gen():
+        yield 1
+        raise Exception("Some error")
```

Comment:
It might be better to use more specialized exception

Reply:
Why is it needed here? It's just test for util function to check what happens when any exception is thrown in the code